switchcraft/data/computercraft/lua/rom/programs/http/pastebin.lua

```diff
@@ -155,15 +165,37 @@ if sCommand == "put" then
   writeCol(colors.lightGray, "Connecting to ")
   writeCol(colors.lime, SC_PASTE_NAME)
   writeCol(colors.lightGray, "... ")
-
-  local response, err, errResponse = http.post(
+  if not tArgs[3] then
+   response, err, errResponse = http.post(
     SC_PASTE_ENDPOINT .. "/api/v1/pastes" ..
     "?language=lua" ..
     "&name=" .. textutils.urlEncode(sName),
     sText,
     { ["Content-Type"] = "text/plain" }
-  )
-
+    )
+  else
```

Comment:
I don't think there's any need to duplicate the whole request flow here just to add some additional arguments. You can still take the body out into a table and serialise it as JSON, but this should be merged into the initial implementation rather than duplicating it entirely.